code/game/turfs/turf.dm

```diff
@@ -445,7 +468,7 @@
 
 	// If we are in a multiz volume and not already inside, we return
 	// the outside value of the highest unenclosed turf in the stack.
-	if((. != OUTSIDE_NO) && HasAbove(z))
```

Comment:
Most commonly, ``. == OUTSIDE_NO`` here because all space and non-planet areas are set to ``is_outside = OUTSIDE_NO`` (see the check above). If the check is left in place, turfs are instantly set to non-outside as soon as their area is changed to a player made or station area, regardless of the presence of a roof. Nothing really behaves as you would expect if this check is left in place.

The downside of removing this check is you're pretty much required to map in a roof (or in space, have the game initialize a roof for you) if the turfs aren't at the top of the Z-stack. I can try to add flexibility to this if you request, but I'm not really sure how to present a roofed turf that doesn't actually have a roof with regards to player mobility and Z-mimic.